src/api/scan/base_controller.go

```diff
@@ -110,118 +117,156 @@ func NewController() Controller {
 	}
 }
 
+func (bc *basicController) collectScanningArtifacts(ctx context.Context, r *scanner.Registration, artifact *ar.Artifact) ([]*ar.Artifact, bool, error) {
+	var (
+		scannable bool
+		artifacts []*ar.Artifact
+	)
+
+	walkFn := func(a *ar.Artifact) error {
+		hasCapability := HasCapability(r, a)
```

Comment:
`HasCapability` is confusing. can we rename it as ArtScannable or sort of?

Reply:
There are two scenarios when artifact is scannable.

1. The scanner has capability for the artifact directly, eg the artifact is docker image.
2. The scanner does not has capability for the artifact, but the artifact is image index and the scanner has capability for any one of the artifacts referenced by the image index.

For these reasons,  rename `HasCapability` to `ArtScannable` is not suitable. And this function will be removed and use `HasCapability` of the scanner directly when this issue https://github.com/goharbor/pluggable-scanner-spec/issues/2 be fixed.